### ISSUE TITLE
User type VariableName doesn't work with immediate_run

### ISSUE THREAD

**Issue by rhuygen** [IvS-KULeuven/gui-executor#58]
We get the following output/error when trying to use a `VariableName()` type with `immediate_run`.

```
New kernel 'python3' started...
Python 3.8.17 (default, Aug 11 2023, 13:56:37) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.3 -- An enhanced Interactive Python. Type '?' for help.

Couldn't load startup script, PYTHONSTARTUP not defined.
----- Running script 'process_model_immediately' in kernel
The code snippet:
 ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 
  response = process_model_immediately()                                                                                                                                                                                                        
 ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 16
     13     return response
     15 if not error:
---> 16     response = main()

Cell In[2], line 10, in main()
      9 def main():
---> 10     response = process_model_immediately()  # [3405691582]
     11     if response is not None:
     12         print(response)

File ~/Documents/PyCharmProjects/gui-executor/src/gui_executor/exec.py:130, in exec_ui.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    127 @wraps(func)
    128 def wrapper(*args, **kwargs):
    129     # you can put extra code to be run here, based on the arguments to exec_ui
--> 130     response = func(*args, **kwargs)
    131     # or here
    132     return response

TypeError: process_model_immediately() missing 1 required positional argument: 'model'
status = 'error'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 16
     13     return response
     15 if not error:
---> 16     response = main()

Cell In[2], line 10, in main()
      9 def main():
---> 10     response = process_model_immediately()  # [3405691582]
     11     if response is not None:
     12         print(response)

File ~/Documents/PyCharmProjects/gui-executor/src/gui_executor/exec.py:130, in exec_ui.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    127 @wraps(func)
    128 def wrapper(*args, **kwargs):
    129     # you can put extra code to be run here, based on the arguments to exec_ui
--> 130     response = func(*args, **kwargs)
    131     # or here
    132     return response

TypeError: process_model_immediately() missing 1 required positional argument: 'model'
----- function 'process_model_immediately' execution finished.
```

**Comment by rhuygen** [comment]
The task that we were executing was:

```
@exec_task(immediate_run=True)
def process_model_immediately(model: VariableName("ldo_model")):
    print(f"Running model immediately")
```

While other positional and keyword arguments are indeed not allowed with `immediate_run`, arguments of type `VariableName` should be allowed.

**Comment by rhuygen** [comment]
Solved in PR #59
